ts/stateTree.ts

```diff
@@ -58,12 +58,21 @@ export class StateTree {
         }
         const leaf = state.toStateLeaf();
         this.stateTree.updateSingle(stateID, leaf);
-        this.states[stateID] = state;
+        // Need to clone the object so that whatever we do on this.states later won't affect the input state.
+        this.states[stateID] = state.clone();
```

Comment:
Fixed a gotcha here. Before this change, the input state would change when we run state transition on states